content/knowledge/consensus_algorithms.md

```diff
@@ -0,0 +1,314 @@
+If more than a treshold of responders vote positively, the transaction is said to collect a `chit`.
+Nodes compute confidence in transactions as the total amount of chits they obtained - they query a transaction only once and rely on new vertices and possible chits added to the progeny to build up their confidence.
+
+TODO: add a picture of avalanche protocol steps
+
+Each of the nodes perform the querying round independently and asynchronously.
+So since one node does not query every other node like in classical consensus, each node performs their own sample of randomly selected nodes, thus able to scale to hundreds of thousands or millions of nodes without adding lots of additional overhead on each node, as they are querying the same number of nodes.
+
+With enough confidence, a decision is finalized immediately.
+This process happens so quickly that Avalanche rivals major payment systems in its ability process and clear transactions in a network.
+
+Avalanche follows the same approach to building a probabilistic consensus algorithm as Nakamoto and goes a few steps further towards a distrubted version of Proof of Stake that scales without the need of a leader election and communication overhead.
+Because it operates without a leader, the system is immune to a large class of attacks that other consensus protocol families face (most importantly Sibyl attack and double spending).
+The large number of validators ensures immutability and censorship resistance that proof-of-work protocols, backed by small numbers of mining pools, or classical proof-of-stake systems cannot achieve.
+
+The consensus supports an arbitrarily large and parametrizable set of adversaries.
+If the network is parameterized for a 33% attacker, and an attacker has 34% stake, unlike with Classical consensus protocols where launching a double spend attack is guaranteed to succeed, with Avalanche it just means they have a slightly higher chance to succeed rather than guaranteed.
+
+## Turning Avalanche into FBA by Making the Algorithm with Locally Defined Validator Set
+
+### Flare System
+
+[Flare](https://drive.google.com/file/d/15adsTes_lhi12PGMxlXfxCsoQsjXvPnS/view) is the first hybrid network that runs a version of Avalanche consensus protocol with Federated byzantine Agreement manner.
+It is integrated with Ethereum Virtual Machine (EVM) and is aims to avoid PoW and PoS mechanisms limiting properties when running its smart contract platform.
+
+Flare Consensus Protocol (FCP) uses FBA and virtual voting (as Hashgraph).
+Flare follows the FBA requirements for reaching netwrok quorum intersection after removing malicious nodes from the system, consequently - quorum interesection is again required between well-behaved nodes.
+
+Unlike Ripple and Stellar, Flare Network runs smart contracts.
+By integrating with EVM, the network can execute Turing complete smart contracts just like dApps in the Ethereum ecosystem.
+
+The protocol relies on two other assumtions.
```

Comment:
```suggestion
The protocol relies on two other assumptions.
```

Reply:
Yep, missed that
Corrected https://github.com/optakt/optakt.github.io/pull/21/commits/f524c8d94d32f6edd6da60a9e7f74a848c44d4ce#diff-4bf8323c6426a93da1ce22be1b8528779ef6b61fdacef3e0d065f1e1dd47bea5R290
Thanks!